sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveStrategies.scala

```diff
@@ -138,6 +138,54 @@ class DetermineTableStats(session: SparkSession) extends Rule[LogicalPlan] {
   }
 }
 
+case class PruneHiveTablePartitions(
```

Comment:
add a todo that we should merge this rule with `PruneFileSourcePartitions`, after we completely make hive a data source.